helix-view/src/gutter.rs

```diff
@@ -1,21 +1,53 @@
 use std::fmt::Write;
+use std::iter::successors;
 
 use crate::{
+    editor::GutterType,
     graphics::{Color, Modifier, Style},
     Document, Editor, Theme, View,
 };
 
+fn count_digits(n: usize) -> usize {
+    successors(Some(n), |&n| (n >= 10).then(|| n / 10)).count()
+}
```

Comment:
Fun snippet :) I'd fully qualify `std::iter::successors` rather than importing since it's only used in one location

Reply:
I'd also add a note that this can be simplified to `n.checked_log10().unwrap_or(0) + 1` if `int_log` gets standardized in stdlib

Reply:
Wow! You all work fast. Stepped away to pet my cat and had a full review when I got back on a PR in draft. Kudos to you!

I'll qualify the use and add a note that this can be done less opaquely once `int_log` is standardized.